### ISSUE TITLE
AbstractCQRSOperation constructor

### ISSUE THREAD

**Issue by clement-hvt** [PrestaShop/PrestaShop#36660]
### Prerequisites

- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.
- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).

### Describe the bug and add attachments

Hi, I think there is a mistake in the constructor of [`AbstractCQRSOperation`](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/ApiPlatform/Metadata/AbstractCQRSOperation.php). The `openapi` constructor parameter allow only boolean type but the inherited class allow `bool|OpenApiOperation|null $openapi = null`.

### Expected behavior

We can set the `openapi` parameter with the class `ApiPlatform\OpenApi\Model\Operation` to specify many things like request body.

### Steps to reproduce

Add an openapi property to an existing ApiPlatform ressource in the module `ps_apiresource`. You can take the example here : [https://api-platform.com/docs/core/openapi/#changing-operations-in-the-openapi-documentation](https://api-platform.com/docs/core/openapi/#changing-operations-in-the-openapi-documentation).

### PrestaShop version(s) where the bug happened

9.0.0

### PHP version(s) where the bug happened

8.1

### Your company or customer's name goes here (if applicable).

Soledis

**Comment by florine2623** [comment]
ping @PrestaShop/tech-council 
maybe @jolelievre this would interest you ^^

**Comment by clement-hvt** [comment]
Hi,

Would it be worth it for me to make a pull request ?

Thanks

**Comment by jolelievre** [comment]
@clement-hvt you're right, I guess since we updated APIPlatform the allowed types evolved since the day this class was create
It's even more types from what I can see it's `bool|OpenApiOperation|Webhook|null`

Don't hesitate to open a PR to fix this, if you are motivated can you check the other parameters as well? `$openapi` may not be the only one that evolved

**Comment by jolelievre** [comment]
Indeed in 3.1 there were less types https://github.com/api-platform/core/blob/3.0/src/Metadata/HttpOperation.php#L100 since we started working on version 2.x it was not updated I guess to make things faster and not check the parameters one by one you can copy all the parameters from the parent constructor so they will all be updated And maybe there are even new fields that are not in our constructor at all and that should be added

Are you ok for making this PR @clement-hvt ? If not I'll look into it later

**Comment by clement-hvt** [comment]
@jolelievre ok i will do it !
